src/vivliostyle/repetitiveelements.js

```diff
@@ -0,0 +1,909 @@
+        /** @const */ this.rootSourceNode = rootSourceNode;
+        /** @type {boolean} */ this.doneInitialLayout = false;
+        /** @type {boolean} */ this.isRoot = false;
+        /** @type {vivliostyle.repetitiveelements.RepetitiveElements} */ this.repetitiveElements = null;
+    };
+    /** @const */ var RepetitiveElementsOwnerFormattingContext = vivliostyle.repetitiveelements.RepetitiveElementsOwnerFormattingContext;
+
+    /**
+     * @override
+     */
+    RepetitiveElementsOwnerFormattingContext.prototype.getName = function() {
+        return "Repetitive elements owner formatting context (vivliostyle.repetitiveelements.RepetitiveElementsOwnerFormattingContext)";
+    };
+
+    /**
+     * @override
+     */
+    RepetitiveElementsOwnerFormattingContext.prototype.isFirstTime = function(nodeContext, firstTime) {
+        return firstTime;
+    };
+
+    /**
+     * @override
+     */
+    RepetitiveElementsOwnerFormattingContext.prototype.getParent = function() {
+        return this.parent;
+    };
+
+    /**
+     * @return {vivliostyle.repetitiveelements.RepetitiveElements}
+     */
+    RepetitiveElementsOwnerFormattingContext.prototype.getRepetitiveElements = function() {
+        return this.repetitiveElements;
+    };
+
+    /**
+     * @param {adapt.vtree.NodeContext} nodeContext
+     * @return {boolean}
+     */
+    RepetitiveElementsOwnerFormattingContext.prototype.isAfterContextOfRootElement = function(nodeContext) {
+        if (nodeContext
+            && nodeContext.formattingContext === this
+            && nodeContext.sourceNode === this.rootSourceNode
+            && nodeContext.after) return true;
+        for (; nodeContext; nodeContext = nodeContext.parent) {
+            if (nodeContext.formattingContext === this) {
+                return false;
+            }
+        }
+        return true;
+    };
+
+    /**
+     * @param {adapt.vtree.NodeContext} position
+     * @returns {Element}
+     */
+    RepetitiveElementsOwnerFormattingContext.prototype.getRootViewNode = function(position) {
+        var root = this.getRootNodeContext(position);
+        return root ? /** @type {Element} */ (root.viewNode) : null;
+    };
+
+    /**
+     * @param {adapt.vtree.NodeContext} nodeContext
+     * @returns {adapt.vtree.NodeContext}
+     */
+    RepetitiveElementsOwnerFormattingContext.prototype.getRootNodeContext = function(nodeContext) {
+        do {
+            if (!this.isInherited(nodeContext)) {
+                return nodeContext;
+            }
+        } while (nodeContext = nodeContext.parent);
+        return null;
+    };
+
+    /**
+     * @param {adapt.vtree.NodeContext} nodeContext
+     * @returns {boolean}
+     */
+    RepetitiveElementsOwnerFormattingContext.prototype.isInherited = function(nodeContext) {
+        return !!(nodeContext
+          && nodeContext.formattingContext === this
+          && nodeContext.sourceNode !== this.rootSourceNode);
+    };
```

Comment:
Fix it in [this commit](https://github.com/vivliostyle/vivliostyle.js/pull/323/commits/3da0763a117cf7736c77add3b42eb995107a1970).